blocks/library/freeform/format-list.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import clickOutside from 'react-click-outside';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { Button, Dashicon } from 'components';
+
+/**
+ * Internal dependencies
+ */
+import './format-list.scss';
+
+class FormatList extends wp.element.Component {
+	constructor() {
+		super( ...arguments );
+		this.switchFormat = this.switchFormat.bind( this );
+		this.toggleMenu = this.toggleMenu.bind( this );
+		this.state = {
+			open: false,
+		};
+	}
+
+	handleClickOutside() {
+		if ( ! this.state.open ) {
+			return;
+		}
+		this.setState( { open: false } );
+	}
+
+	toggleMenu() {
+		this.setState( {
+			open: ! this.state.open,
+		} );
+	}
+
+	switchFormat( newValue ) {
+		if ( this.props.value !== newValue && this.props.onFormatChange ) {
+			this.props.onFormatChange( newValue );
+		}
+		this.setState( { open: false } );
+	}
+
+	render() {
+		const { formats = [ { text: '', value: '', textStyle: () => '' } ] } = this.props;
+		const selectedValue = this.props.value;
+		return (
+			<div className="editor-format-list">
+				<Button
+					className="editor-format-list__toggle"
+					onClick={ this.toggleMenu }
+					aria-haspopup="true"
+					aria-expanded={ this.state.open }
+					aria-label={ wp.i18n.__( 'Change format' ) }
+				>
+					<div className="formats">
+						{ formats.map( ( { text, value } ) => (
```

Comment:
Well they're there so the dropdown stays the size of the largest item.

I am stacking the lines on top of each other while hiding all but the active item. This results in the width of the button matching the largest item which appears to be how it was in the mockup.

I am happy to take suggestions on how to achieve this effect in another way.